component/registry/registry.go

```diff
@@ -0,0 +1,63 @@
+package registry
```

Comment:
I agree, I got confused a bit initially. Maybe using `store` like we do elsewhere may be better?